### PR TITLE
[PRO-235] Feature: name accounts

### PR DESCRIPTION
**Pull request by Ivan-Mahda**
## Purpose

As a browser wallet user, I want the option to name my accounts.

## Changes

Option to edit account name
When Edit icon clicked, User able to edit account name. Name saved in local storage
Changed name displayed across all BrowserWallet

Screenshot 2024-05-13 at 18 00 45
Screenshot 2024-05-13 at 18 01 05

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [ ] (If necessary) I have updated the CHANGELOG.

By submitting the contribution I accept the terms and conditions of the
Contributor License Agreement v1.0
- link: https://developers.concordium.com/CLAs/Contributor-License-Agreement-v1.0.pdf

- [ ] I accept the above linked CLA.

### CODE REVIEW

File: packages/browser-wallet/src/popup/page-layouts/MainLayout/AccountList/AccountList.tsx [hunk 1]
```diff
@@ -1,21 +1,83 @@
...
+            setIsEditing(false);
+        }
+    }, [name]);
+
+    const keyHandlerEnter: KeyboardEventHandler<HTMLInputElement> = (e) => {
+        if (e.key === 'Enter') {
+            e.preventDefault();
+            handleSubmitName();
+        }
+    };
```
Comment (limemloh) [inline packages/browser-wallet/src/popup/page-layouts/MainLayout/AccountList/AccountList.tsx:45]: The input seems to be flickering when I type into it, can we avoid it being rerendered on every key stroke?

File: packages/browser-wallet/src/popup/page-layouts/MainLayout/AccountList/AccountList.tsx [hunk 2]
```diff
@@ -1,21 +1,83 @@
...
+} from '@popup/shared/utils/account-helpers';
 import { WalletCredential } from '@shared/storage/types';
 import { useAccountInfo } from '@popup/shared/AccountInfoListenerContext';
-import { isDelegatorAccount, isBakerAccount, AccountInfo } from '@concordium/web-sdk';
+import { AccountInfo, isBakerAccount, isDelegatorAccount } from '@concordium/web-sdk';
+import IconButton from '@popup/shared/IconButton';
+import { InlineInput } from '@popup/shared/Form/InlineInput';
 import EntityList from '../EntityList';
 
+const ACCOUNT_NAME_MAX_LENGTH = 12;
```
Comment (soerenbf) [inline packages/browser-wallet/src/popup/page-layouts/MainLayout/AccountList/AccountList.tsx:26]: What's the reason behind this max length? I could see people wanting to name their accounts something longer than this 🙂

**Review by soerenbf — APPROVED** [review]
One concern, otherwise looks good 😄

**Review by limemloh — APPROVED** [review]

File: packages/browser-wallet/src/popup/page-layouts/MainLayout/AccountList/AccountList.tsx (hunk 2 above)
Comment (Ivan-Mahda) [reply packages/browser-wallet/src/popup/page-layouts/MainLayout/AccountList/AccountList.tsx]: Places where the account name is used, have been designed for the shortened account address. With this limitation it won't break existing design.

**Pull request merged**